packages/serverpod_serialization/lib/src/copy_with.dart

```diff
@@ -68,13 +50,10 @@ dynamic _guardedCopyWith(dynamic element) {
   if (element is Enum) return element;
 
   // Required as the extension with clone() is not found otherwise.
-  if (element is ByteData) return element.clone();
-  if (element is List) return element.clone();
-  if (element is Map) return element.clone();
-
-  try {
-    return element.copyWith();
-  } on NoSuchMethodError {
-    throw 'No copyWith method found on ${element.runtimeType}';
+  if (element is ByteData) return element.clone() as T;
+  if (element is List || element is Map) {
+    throw UnimplementedError("Can't clone `List` or `Map` types");
   }
```

Comment:
Suggestion: Use `UnsupportedError` instead.

Since the type is actually not supported by the method. Could also explain why these can not be cloned by this method. `"List" and "Map" are not shallow types`.